src/popup/components/InfiniteScroll.vue

```diff
@@ -0,0 +1,49 @@
+<template>
+  <div
+    class="infinite-scroll"
+    @scroll="onViewportScroll"
```

Comment:
In my opinion this is wrong - we should not watch the scroll here. We have the scroll events attached to the viewport with the composable.